core/kazoo_number_manager-1.0.0/src/knm_phone_number.erl

```diff
@@ -0,0 +1,549 @@
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec reserve_history(number()) -> ne_binaries().
+reserve_history(Number) ->
+    Number#number.reserve_history.
+
+-spec set_reserve_history(number(), ne_binaries()) -> number().
+set_reserve_history(N, History) ->
+    N#number{reserve_history=History}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec ported_in(number()) -> boolean().
+ported_in(Number) ->
+    Number#number.ported_in.
+
+-spec set_ported_in(number(), boolean()) -> number().
+set_ported_in(N, Ported) ->
+    N#number{ported_in=Ported}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec module_name(number()) -> ne_binary().
+module_name(Number) ->
+    Number#number.module_name.
+
+-spec set_module_name(number(), ne_binary()) -> number().
+set_module_name(N, Name) ->
+    N#number{module_name=Name}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec region(number()) -> ne_binary().
+region(Number) ->
+    Number#number.region.
+
+-spec set_region(number(), ne_binary()) -> number().
+set_region(N, Region) ->
+    N#number{region=Region}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec auth_by(number()) -> ne_binary().
+auth_by(Number) ->
+    Number#number.auth_by.
+
+-spec set_auth_by(number(), ne_binary()) -> number().
+set_auth_by(N, AuthBy) ->
+    N#number{auth_by=AuthBy}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec dry_run(number()) -> ne_binary().
+dry_run(Number) ->
+    Number#number.dry_run.
+
+-spec set_dry_run(number(), ne_binary()) -> number().
+set_dry_run(N, DryRun) ->
+    N#number{dry_run=DryRun}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec doc(number()) -> wh_json:object().
+doc(Number) ->
+    Number#number.doc.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec routines(number(), [function()]) -> number_return().
+routines(Number, Routines) ->
```

Comment:
This function sounds like a getter, as in 'get all the routines'. Maybe `execute_routines` or `execute_funs`, something that gives the reader a better idea of what the function does.